cluster/src/main/java/io/atomix/cluster/GroupMembershipConfig.java

```diff
@@ -25,10 +26,45 @@
   private static final int DEFAULT_FAILURE_TIMEOUT = 10000;
   private static final int DEFAULT_PHI_FAILURE_THRESHOLD = 10;
 
+  private MemberLocationProvider locationProvider;
   private int heartbeatInterval = DEFAULT_HEARTBEAT_INTERVAL;
   private int phiFailureThreshold = DEFAULT_PHI_FAILURE_THRESHOLD;
   private int failureTimeout = DEFAULT_FAILURE_TIMEOUT;
 
+  /**
+   * Returns the member location provider.
+   *
+   * @return the member location provider
+   */
+  public MemberLocationProvider getLocationProvider() {
+    return locationProvider;
+  }
+
+  /**
+   * Sets the member location provider.
+   *
+   * @param locationProvider the member location provider
+   * @return the membership configuration
+   */
+  public GroupMembershipConfig setLocationProvider(Class<? extends MemberLocationProvider> locationProvider) {
```

Comment:
I wonder why do you use reflexion here while the object is just passed on the `Builder` class.

Reply:
It's for configuration files. The configuration parser will apply a class name to this configuration when read from a file.